main.rb

```diff
@@ -0,0 +1,130 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    to_enum(:my_each) unless block_given?
```

Comment:
If you change this line from `to_enum(:my_each)` to `return to_enum` for all four methods issues with Enumerator will be fixed.